### PR TITLE
Fix: align V4.1 references and tighten C1A precision checks

### PR DESCRIPTION
**Pull request by high-cloud**
- Align FP32 projection accumulation and BF16 prefill indexer rounding
  with the reference computation.
- Mirror tiled BF16 PV and preserve FP32 partials through TP reduction.
- Validate Top-K eligibility, cutoff quality, order, and padding before
  checking output against reference caches and original input weights
  for an accepted alternative selection.
- Replace C1A prefill's fixed absolute output tolerance with per-row
  RMS error <= 1e-6 + 1% of reference RMS and peak error <= 1e-5 + 5% of
  reference RMS. Reject non-finite values without a global outlier quota.
- Order index-key decode after weight projection in the shared C1A
  Full/Reindex prefill and decode helpers. Document this temporary A5
  C2V UB-overlap workaround and its parallelism cost, with an upstream
  issue reference in both kernels.
- Document the reference arithmetic and validation contracts.

Upstream runtime/ISA issue: https://github.com/hw-native-sys/pypto/issues/2829

### CODE REVIEW

File: models/deepseek_v4_1_flash/golden.py [hunk 1]
```diff
@@ -243,7 +244,7 @@ def paged_indexer(
     q = mxfp8_linear(qr, wq_b, wq_b_scale).unflatten(-1, (index_heads, index_dim))
     rd = cos.shape[-1] * 2
     q = torch.cat((q[..., :-rd], rope_interleave(q[..., -rd:], cos, sin)), dim=-1)
-    weights = torch.matmul(x, weights_proj)
+    weights = torch.matmul(x.float(), weights_proj.float())
     weights = weights * index_dim**-0.5 * index_heads**-0.5
     scores = torch.einsum("thd,tkd->thk", q.float(), keys.float()).relu()
     scores = (scores * weights.float().unsqueeze(-1)).sum(dim=-2)
```
[the remark this answers is not in the document]
Comment (high-cloud) [reply models/deepseek_v4_1_flash/golden.py]: Implemented in 3b22e53. The prefill C1A scorer and paged-indexer golden now round projected/scaled index weights, QK dots, weighted products, and head reductions to BF16, while retaining FP32 accumulation and FP32 storage for the rounded Top-K scores. Compressor, index-key, and grouped-output accumulation changes are preserved. Scalar-reference regression tests pass, and captured weights/scores match the golden bit-for-bit on all four ranks. The intermittent TP4 Top-K mismatch is tracked separately and is not claimed fixed by this arithmetic change.

**Comment by high-cloud** [comment]
Added the temporary C2V ordering workaround in f0a6f20 and filed
https://github.com/hw-native-sys/pypto/issues/2829 with the complete standalone
reproducer (PyPTO + PyTorch only). The final single-card script fails 5/5
unordered runs and passes 5/5 ordered controls at zero tolerance. The upstream
runtime/ISA defect remains open; this PR only avoids the observed overlap.

Validation on the unchanged CI-pinned A5 environment (PyPTO 1deae896,
simpler 39ce891d, PTOAS 0.61, PTO ISA 5a4f74cb, CANN 9.2.0):

| Case | Result |
| --- | --- |
| Prefill Reindex, TP4, default mixed fixture, one level-4 + two ordinary runs | 3/3 PASS |
| Prefill Full, TP4, default mixed fixture | PASS |
| Prefill Full, TP4, causal 2 tokens | PASS |
| Decode Reindex, TP4, 3 ragged tokens, level-4 capture | PASS |
| Decode Full, TP4, 3 ragged tokens | PASS |
| Prefill Full, TP1, causal 2 tokens | PASS |
| Golden unit tests | 338 PASS |
| All repository pre-commit hooks | PASS |

The fresh prefill and decode graphs each show the direct weight-projection
completion -> key-decoder edge on all four ranks. Prefill output uses the
existing strict RMS/peak gate; neither arithmetic nor thresholds changed in
this workaround.

Indexer source audit:

- V4.1 prefill Full/Reindex share the reproduced failing helper. Decode
  Full/Reindex have the same independently scheduled mixed-weight/key-decoder
  pattern and now receive the guard too. Decode was not independently shown
  to fail before the guard.
- V4.1 Reuse does not execute these indexer tasks; the hierarchical selector
  itself is Vector-only.
- V4 Pro, Flash MTP, and DSpark decode weight projections store FP32 partials
  to GM and reduce them in a separate Vector task. They do not have this exact
  mixed weight-epilogue/FP4 decoder pair.
- V4 Pro prefill already joins weight/query completion before its compressor.
  Flash MTP and DSpark prefill use mixed weight projections and INT8 cache
  scoring rather than the separate V4.1 packed-FP4 decoder. General A5 C2V
  overlap risk is not ruled out; no failures were reproduced or changes made
  in those models during this audit.
- GLM5.3 Flash projection/score bodies are unimplemented stubs.

The other models above were inspected, not device-certified by this change.
